models/routemodels/route.go

```diff
@@ -13,38 +14,39 @@ const (
 )
 
 type Route struct {
-	ID         string    `json:"id,omitempty" bson:"id,omitempty"`
-	Type       RouteType `json:"type,omitempty" bson:"type,omitempty"`
-	WebhookURL string    `json:"webhook_url,omitempty" bson:"webhook_url,omitempty"`
+	ID         string    `json:"id,omitempty" bson:"id,omitempty" msgpack:",omitempty"`
+	Type       RouteType `json:"type,omitempty" bson:"type,omitempty" msgpack:",omitempty"`
+	WebhookURL string    `json:"webhook_url,omitempty" bson:"webhook_url,omitempty" msgpack:",omitempty"`
 }
 
-func (r *Route) Valid() error {
-	if r.ID == "" {
-		return fmt.Errorf("invalid route id")
+var (
+	errInvalidRouteID    = errors.New("invalid route id")
+	errInvalidRouteType  = errors.New("invalid route type. Only REST is allowed")
+	errInvalidWebhookURL = errors.New("invalid webhook url")
+)
+
+func (r Route) Valid() error {
+	if len(r.ID) == 0 {
+		return errInvalidRouteID
 	}
 	if r.Type != REST {
-		return fmt.Errorf("invalid route type. Only REST is allowed")
+		return errInvalidRouteType
 	}
-	if r.WebhookURL == "" {
-		return fmt.Errorf("invalid webhook url")
+	if len(r.WebhookURL) == 0 {
+		return errInvalidWebhookURL
 	}
-
 	return nil
 }
 
-// TODO: Change to msgpack later
-func (r *Route) ToBytes() ([]byte, error) {
-	return json.Marshal(&r)
+func (r Route) ToBytes() ([]byte, error) {
+	encodedData, err := msgpack.Marshal(&r)
+	return encodedData, err
 }
 
-// GetRouteFromBytes returns the route struct from byte array
-// TODO: Change to msgpack later
-func GetRouteFromBytes(by []byte) (*Route, error) {
-	var r Route
-	err := json.Unmarshal(by, &r)
-	if err != nil {
-		return nil, err
+func GetRouteFromBytes(encodedData []byte, r *Route) error {
```

Comment:
Injecting a pointer to route instead of returning it helps in saving short term heap allocations